ansible/configs/ansible-multitier-infra/pre_infra.yml

```diff
@@ -11,20 +11,26 @@
     - debug:
         msg: "Step 000 Pre Infrastructure"
 
-    - name: Generate SSH keys
+    - when: set_env_authorized_key | bool
```

Comment:
Have a look at this role. It already covers these 3 tasks I believe.

https://github.com/redhat-cop/agnosticd/blob/development/ansible/roles-infra/infra-local-create-ssh_key/tasks/main.yml